Add a small fastpath test for native mha

zrphercule opened the pull request:
Summary: We dont have a small fast path passing test for mha before, this diff added one for better testing

Test Plan: buck build mode/dev-nosan -c fbcode.platform=platform009 -c fbcode.enable_gpu_sections=true caffe2/test:nn && buck-out/dev/gen/caffe2/test/nn\#binary.par -r test_multihead_attn_fast_path_small_test

Differential Revision: D37834319

zrphercule commented:
@pytorchbot merge -g

erichan1 reviewed (approved):
Thanks for adding the test!

zrphercule commented:
@pytorchbot merge -g

zrphercule commented:
@pytorchbot merge

zrphercule2 committed:
Add a small fastpath test for native mha (#81432)

Summary:
We dont have a small fast path passing test for mha before, this diff added one for better testing

Pull Request resolved: https://github.com/pytorch/pytorch/pull/81432

Test Plan: buck build mode/dev-nosan -c fbcode.platform=platform009 -c fbcode.enable_gpu_sections=true caffe2/test:nn && buck-out/dev/gen/caffe2/test/nn\#binary.par -r test_multihead_attn_fast_path_small_test

Reviewed By: erichan1, mikekgfb

Differential Revision: D37834319

Pulled By: zrphercule

fbshipit-source-id: 832a67a02322c88390bacd65dc7c287750db65ee

zrphercule commented:
@pytorchbot merge -g
